syscalls/gen-syscall-list/build.rs

```diff
@@ -14,29 +14,71 @@ use {
  * to verify undefined symbols in a .so module that cargo-build-sbf has built.
  */
 fn main() {
-    let syscalls_rs_path = PathBuf::from("../src/lib.rs");
-    let syscalls_txt_path = PathBuf::from("../../platform-tools-sdk/sbf/syscalls.txt");
+    let syscalls_rs_name = "../src/lib.rs";
+    let syscalls_txt_name = "../../platform-tools-sdk/sbf/syscalls.txt";
+    let build_sbf_syscalls_name = "../../platform-tools-sdk/cargo-build-sbf/src/syscalls.rs";
+    println!("cargo::rerun-if-changed={syscalls_rs_name}");
+    println!("cargo::rerun-if-changed={syscalls_txt_name}");
+    println!("cargo::rerun-if-changed={build_sbf_syscalls_name}");
+    println!("cargo::rerun-if-changed=build.rs");
+
+    let syscalls_rs_path = PathBuf::from(syscalls_rs_name);
+    let syscalls_txt_path = PathBuf::from(syscalls_txt_name);
+    let build_sbf_syscalls_path = PathBuf::from(build_sbf_syscalls_name);
     println!(
-        "cargo:warning=(not a warning) Generating {1} from {0}",
+        "cargo::warning=(not a warning) Generating {1} and {2} from {0}",
         syscalls_rs_path.display(),
-        syscalls_txt_path.display()
+        syscalls_txt_path.display(),
+        build_sbf_syscalls_path.display(),
     );
 
+    let old_num_syscalls = File::open(&syscalls_txt_path)
+        .map(|file| {
+            let reader = BufReader::new(file);
+            reader.lines().count()
+        })
+        .unwrap_or(0);
```

Comment:
Since we now have `platform-tools-sdk/cargo-build-sbf/src/syscalls.rs` and in #9348 you are not using syscalls.txt anymore, could we read from syscalls.rs instead?